### PR TITLE
Allow non-Az modules in requirements.psd1

### PR DESCRIPTION
**Pull request by AnatoliB**
Allow non-Az modules in requirements.psd1

Addresses https://github.com/Azure/azure-functions-powershell-worker/issues/246

### CODE REVIEW

File: docs/designs/PowerShell-AzF-Overall-Design.md [hunk 1]
```diff
@@ -466,10 +466,13 @@ Note that, checking out a PowerShell Manager instance from the pool is a blockin
 
 The goal is to let the user declare the dependencies required by functions, and rely on the service automatically locating and installing the dependencies from the PowerShell Gallery or other sources, taking care of selecting the proper versions, and automatically upgrading the dependencies to the latest versions (if allowed by the version specifications provided by the user).
 
-Dependencies are declared in the _requirements.psd1_ file (_manifest_) as a collection of pairs (<_name_>, <_version specification_>). Currently, only _Az_ module is allowed, and the version specification should strictly match the following pattern: `<major version>.*`, so a typical manifest looks like this:
+Dependencies are declared in the _requirements.psd1_ file (_manifest_) as a collection of pairs (<_name_>, <_version specification_>). Currently, the version specification should strictly match the following pattern: `<major version>.*`, so a typical manifest looks like this:
```
[the remark this answers is not in the document]
Comment (AnatoliB) [reply docs/designs/PowerShell-AzF-Overall-Design.md]: Thanks, done.

File: src/DependencyManagement/DependencyManifest.cs [hunk 2]
```diff
@@ -90,27 +87,23 @@ private Hashtable ParsePowerShellDataFile()
                 throw new ArgumentException(errorMsg);
             }
 
+            if (hashtable.Count > _maxDependencyEntries)
+            {
+                var message = string.Format(PowerShellWorkerStrings.TooManyDependencies, RequirementsPsd1FileName, hashtable.Count, _maxDependencyEntries);
```
[the remark this answers is not in the document]
Comment (AnatoliB) [reply src/DependencyManagement/DependencyManifest.cs]: Thanks, I like your version more (just removing the extra parentheses and also moving the word "supported": `The number of entries in the 'Requirements.psd1' file is 11, which exceeds the maximum supported number of entries (10).`

**Comment by dfinke** [comment]
Is there a way to determine if this feature is deployed, short of trying it?

**Comment by Francisco-Gamino** [comment]
Hello @dfinke, this will be part of the next deployment which starts mid next week. This will be available in Azure by mid August.

**Comment by AnatoliB** [comment]
@dfinke Technically, there is a way to track this change merging into the master branch, then getting into the next PowerShell worker release, then into the Functions host release, etc. However, this can get complicated. Trying this out is definitely easier. Mid August is a good estimate, and we will also update this thread when the deployment is done.

**Pull request merged**